assets/src/design-system/components/input/index.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import propTypes from 'prop-types';
+import { useMemo } from 'react';
+import styled, { css } from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+/**
+ * Internal dependencies
+ */
+import { Text } from '..';
+import { themeHelpers, THEME_CONSTANTS } from '../../theme';
+
+const Container = styled.div`
+  margin: 0 12px;
+`;
+
+const Label = styled.label(
+  ({ disabled, theme }) =>
+    disabled &&
+    css`
+      ${Text} {
+        color: ${theme.colors.fg.disable};
+      }
+    `
+);
```

Comment:
Thoughts on having a `Label` component to use for all the labels of form components?

Reply:
Great idea! I think that would be rad

Reply:
Added it as a state on the `<Text />` component